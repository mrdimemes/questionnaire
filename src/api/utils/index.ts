export { default as getFetchError } from "./getFetchError";