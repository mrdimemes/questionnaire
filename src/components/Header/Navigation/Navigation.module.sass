@use "src/sass/foundation/mixins"

.body
  display: flex
  align-items: center
  column-gap: 30px

.link
  display: block
  @include mixins.unhighlightable

.active
  font-weight: 800