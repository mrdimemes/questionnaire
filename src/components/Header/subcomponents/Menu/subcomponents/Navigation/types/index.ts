export type { NavigationProps } from "./NavigationProps";