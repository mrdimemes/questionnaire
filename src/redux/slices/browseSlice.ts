import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export enum NavigationPage {
  Main = "MAIN",
  Tags = "TAGS",
  QuestionnaireList = "QUESTIONNAIRES",
  None = "NONE"
}

const initialState = {
  activePage: NavigationPage.None
}

export const browseSlice = createSlice({
  name: 'browse',
  initialState,
  reducers: {
    setActivePage: (state, action: PayloadAction<NavigationPage>) => {
      state.activePage = action.payload;
    }
  },
})

export const { setActivePage } = browseSlice.actions;
export default browseSlice.reducer;