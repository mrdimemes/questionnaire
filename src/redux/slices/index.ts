export { default as authSlice } from "./authSlice";
export { default as browseSlice } from "./browseSlice";
export { default as tagsSlice } from "./tagsSlice";
export { default as themeSlice } from "./themeSlice";