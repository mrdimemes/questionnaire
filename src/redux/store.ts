import { combineReducers, configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from '@reduxjs/toolkit'
import { authSlice, tagsSlice, browseSlice, themeSlice } from "./slices";

const rootReducer = combineReducers({
  tags: tagsSlice,
  browse: browseSlice,
  auth: authSlice,
  theme: themeSlice
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState
  })
}

const store = setupStore();

export default store;
export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore['dispatch'];