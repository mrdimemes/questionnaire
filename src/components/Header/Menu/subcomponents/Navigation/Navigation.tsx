import classNames from "classnames";
import { useActivePageSelector } from "src/redux/hooks";
import { NavigationPage } from "src/redux/slices/browseSlice";
import { Link } from "react-router-dom";

import styles from "./Navigation.module.sass";

import type { NavigationProps } from "./types";


const Navigation = ({ onClick }: NavigationProps) => {
  const activePage = useActivePageSelector();

  return (
    <nav className={styles.Navigation}>

      <Link
        to="/questionnaire-client"
        className={classNames(
          styles.link,
          { [styles.active]: activePage === NavigationPage.Main },
        )}
        onClick={onClick}
      >
        Главная
      </Link>

      <Link
        to="/questionnaire-client/questionnaires"
        className={classNames(
          styles.link,
          { [styles.active]: activePage === NavigationPage.QuestionnaireList },
        )}
        onClick={onClick}
      >
        Опросы
      </Link>

      <Link
        to="/questionnaire-client/tags"
        className={classNames(
          styles.link,
          { [styles.active]: activePage === NavigationPage.Tags },
        )}
        onClick={onClick}
      >
        Тэги
      </Link>

    </nav>
  );
};

export default Navigation;