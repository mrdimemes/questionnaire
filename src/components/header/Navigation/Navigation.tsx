import classNames from "classnames";
import { useAppSelector } from "src/redux/hooks";
import { NavigationPage } from "src/redux/slices/browseSlice";
import { Link } from "react-router-dom";
import styles from "./Navigation.module.sass";


const Navigation = () => {
  const activePage = useAppSelector((state) => state.browse.activePage);

  return (
    <nav className={styles.body}>

      <Link to="/" className={classNames(
        styles.link,
        { [styles.active]: activePage === NavigationPage.Main }
      )}>Главная</Link>

      <Link to="/questionnaires" className={classNames(
        styles.link,
        { [styles.active]: activePage === NavigationPage.QuestionnaireList }
      )}>Опросы</Link>

      <Link to="/tags" className={classNames(
        styles.link,
        { [styles.active]: activePage === NavigationPage.Tags }
      )}>Тэги</Link>

    </nav>
  )
}

export default Navigation;