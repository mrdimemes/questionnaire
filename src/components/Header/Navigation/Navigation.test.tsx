import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderWithProvidersAndRouters } from "src/utils/test-utils";
import { setupStore } from "src/redux/store";
import { NavigationPage } from "src/redux/slices/browseSlice";
import Navigation from "./Navigation";

describe("Header/Navigation component", () => {
  test("Navigation renders", () => {
    renderWithProvidersAndRouters(<Navigation />);
    expect(screen.getByText(/главная/i)).toBeInTheDocument();
    expect(screen.getByText(/тэги/i)).toBeInTheDocument();
    expect(screen.getByText(/опросы/i)).toBeInTheDocument();
  });

  test("Navigation snapshot", () => {
    const { container } = renderWithProvidersAndRouters(<Navigation />);
    expect(container).toMatchSnapshot();
  });
});