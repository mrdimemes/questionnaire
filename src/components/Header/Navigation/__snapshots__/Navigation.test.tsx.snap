// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header/Navigation component Navigation snapshot 1`] = `
<div>
  <nav
    class="body"
  >
    <a
      class="link"
      href="/"
    >
      Главная
    </a>
    <a
      class="link"
      href="/questionnaires"
    >
      Опросы
    </a>
    <a
      class="link"
      href="/tags"
    >
      Тэги
    </a>
  </nav>
</div>
`;
